### PR TITLE
Add wlrlui -m to automatically match profiles

### PR DESCRIPTION
**Pull request by levnikmyskin**
Hi,
This PR adds an `-m` option to `wlrlui` to automatically match profiles.
Basically, it will iterate over all saved profiles (sorted alphabetically), and it will apply the first profile where all currently connected monitors `uid` match.

This can also be used together with any script that listens for monitor changes, in order to apply the correct profile every time a monitor is connected/disconnected.
For instance, using [pyprland monitors](https://github.com/hyprland-community/pyprland/wiki/monitors), this should work, despite a little cumbersome:
```toml
[monitors.hotplug_commands]
# add all your monitors
"DELL P2417H CJFH277Q3HCB" = "wlrlui -m"
"Sony Something" = "wlrlui -m"
```

### CODE REVIEW

**Review by fdev31 — COMMENTED** [review]
I don't get why it's not prof_uids == current_uids instead ...

**Comment by fdev31** [comment]
I love the idea!

File: src/wlr_layout_ui/__init__.py [hunk 1]
```diff
@@ -26,45 +51,38 @@ def main():
             print("")
             for p in profiles:
                 print(f" - {p}")
+        elif sys.argv[1] == "-m":
+            load()
+            current_uids = set(di.uid for di in displayInfo)
+            for key in sorted(profiles.keys()):
```
Comment (fdev31) [inline src/wlr_layout_ui/__init__.py]: I think the .keys() is superfluous

File: src/wlr_layout_ui/__init__.py [hunk 2]
```diff
@@ -26,45 +51,38 @@ def main():
...
+                prof_uids = {p["uid"] for p in profiles[key]}
+                # check that the two sets have the same elements
+                if len(prof_uids & current_uids) == len(current_uids):
```
Comment (fdev31) [inline src/wlr_layout_ui/__init__.py]: shouldn't be `prof_uids == current_uids` ?

File: src/wlr_layout_ui/__init__.py [hunk 3]
```diff
@@ -26,45 +51,38 @@ def main():
...
+                    print(f"Matched profile {key}. Applying it...")
+                    apply_profile(profiles[key])
+                    sys.exit(0)
```
Comment (fdev31) [inline src/wlr_layout_ui/__init__.py:63]: maybe just a return here...

File: src/wlr_layout_ui/__init__.py [hunk 4]
```diff
@@ -26,45 +51,38 @@ def main():
...
+            print(f"No profile found: {sys.argv[1]}")
+            sys.exit(1)
+
         elif sys.argv[1][0] == "-":
             load()
             print(
                 """With no options, launches the GUI
 Options:
              -l : list profiles
+             -m : find a profile that matches the current displays layout, and apply it.
```
Comment (fdev31) [inline src/wlr_layout_ui/__init__.py]: matches the current displays layout => matches the currently plugged displays set or something like this (it doesn't try to match the "layout", is it?)

**Review by fdev31 — CHANGES_REQUESTED** [review]
Almost ready for merge, most important clarification is about the "main" condition of the code

**Comment by fdev31** [comment]
> This can also be used together with any script that listens for monitor changes, in order to apply the correct profile every time a monitor is connected/disconnected. For instance, using [pyprland monitors](https://github.com/hyprland-community/pyprland/wiki/monitors), this should work, despite a little cumbersome:
> 
> ```toml
> [monitors.hotplug_commands]
> # add all your monitors
> "DELL P2417H CJFH277Q3HCB" = "wlrlui -m"
> "Sony Something" = "wlrlui -m"
> ```

Any suggestion here? something like what follows?

```toml
[monitors]
hotplug_command = "wlrlui -m" 
```

File: src/wlr_layout_ui/__init__.py (hunk 2 above)
Comment (levnikmyskin) [reply src/wlr_layout_ui/__init__.py]: True! I was thinking it in terms of set comparisons, and just went with what felt the most natural way of comparing sets...but that's a nice shortcut :)

**Comment by levnikmyskin** [comment]
Implemented the changes as requested.

> Any suggestion here? something like what follows?

Yes that would work!

**Comment by fdev31** [comment]
Great! I'm ok to add this to pyprland, if I forget you can open a feature request (even one-liner linking here, it's just a reminder)

Thank you for the contribution!

**Comment by levnikmyskin** [comment]
> Great! I'm ok to add this to pyprland, if I forget you can open a feature request (even one-liner linking here, it's just a reminder)
> 
> Thank you for the contribution!

Theoretically that should work already (with `unknown`), if no monitors are configured in pyprland right?

**Comment by fdev31** [comment]
I think I removed it at some point because it was not so straightforward in the end...

**Pull request merged**